lusidtools/commands/commands.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import sys
+import lusidtools.lpt as lpt
+import os
+import ast
+
+
+def find_tools():
+    root = lpt.__path__[0]
+    for filename in os.listdir(root):
+        if filename.endswith('.py'):
+            with open(os.path.join(root, filename), "rt") as file:
+                toolname = filename[:-3]
+                toolmodule = toolname
+                tooltip = ""
+                tree = ast.parse(file.read(), filename=filename)
+                for f in tree.body:
+                    if isinstance(f, ast.Assign):
+                        if (len(f.targets) == 1 and
+                                isinstance(f.targets[0], ast.Name) and
+                                isinstance(f.value, ast.Str)):
+                            if f.targets[0].id == "TOOLNAME":
+                                toolname = f.value.s
+                            elif f.targets[0].id == "TOOLTIP":
+                                tooltip = f.value.s
+                    if isinstance(f, ast.FunctionDef):
+                        if f.name == 'main':
+                            yield toolname, (toolmodule, tooltip)
```

Comment:
This is fascinating. I've never seen this before. I understand what you are doing here and it is really nice. Can we limit the nesting to make it easier to understand? Anymore than 2 levels of nesting starts to get a little dicey and we are already a level deep here if you make the change above.